tensorrt_llm/_torch/modules/attention.py

```diff
@@ -541,25 +541,41 @@ def forward(
             q_ctx = q[:num_ctx_tokens, ...]
             compressed_kv_ctx = compressed_kv[:num_ctx_tokens, ...]
             k_pe_ctx = k_pe[:num_ctx_tokens, ...]
+            q_ctx, latent_cache_ctx = maybe_execute_in_parallel(
+                lambda: self.q_b_proj(q_ctx),
+                lambda: torch.concat([compressed_kv_ctx, k_pe_ctx], dim=-1),
+                self.ln_events[0],
+                self.ln_events[1],
+                self.aux_stream,
+            )
```

Comment:
Instead of split and then do two gemm/concat, you can do gemm/concat and then split the latent_cache and pass them separately to context/gen.